Decred Wallet/Custom Views/FloatingLabelInputs/FloatingPlaceholderLabel.swift

```diff
@@ -0,0 +1,109 @@
+    @IBInspectable var rightPadding: CGFloat = 4
+
+    var topConstraint: NSLayoutConstraint?
+
+    let normalColor: UIColor = UIColor.appColors.lightBluishGray
+    var activeColor: UIColor = UIColor.appColors.lightBlue
+    let normalFontSize: CGFloat = 16
+    let smallerFontSize: CGFloat = 14
+
+    override var text: String? {
+        didSet {
+            // set position after the text did set, since the label position calculated in real-time using the height of the label
+            self.setPosition()
+        }
+    }
+
+    override init(frame: CGRect) {
+        super.init(frame: frame)
+        self.initView()
+    }
+
+    required init?(coder aDecoder: NSCoder) {
+        super.init(coder: aDecoder)
+        self.initView()
+    }
+
+    public func initView() {
+        self.backgroundColor = UIColor.white
+        self.clipsToBounds = true
+        self.translatesAutoresizingMaskIntoConstraints = false
+        self.textColor = self.normalColor
+        self.setPosition()
+    }
+    
+    private func setPosition() {
+        guard let floatingLabelInput = superview as? FloatingLabelProtocol else { return }
+
+        self.font = self.font.withSize(floatingLabelInput.isParentEmpty() ? self.normalFontSize : self.smallerFontSize)
+        self.sizeToFit()
+        let calculatedTopConstraintValue = floatingLabelInput.isParentEmpty() ? (self.superview!.frame.size.height - self.intrinsicContentSize.height) / 2 : self.intrinsicContentSize.height / -2
+
+        if self.topConstraint == nil {
+            self.topConstraint = self.topAnchor.constraint(equalTo: self.superview!.topAnchor, constant: calculatedTopConstraintValue)
+            NSLayoutConstraint.activate([
+                self.leadingAnchor.constraint(equalTo: self.superview!.leadingAnchor, constant: 16),
+                self.topConstraint!
+            ])
+        } else {
+            self.topConstraint?.constant = calculatedTopConstraintValue
+        }
+    }
+
+    func parentEditingDidBegin() {
+        UIView.animate(withDuration: 0.1) {
+            self.textColor = self.activeColor
+            self.font = self.font.withSize(self.smallerFontSize)
+            self.topConstraint?.constant = self.intrinsicContentSize.height / -2
+            self.superview?.layoutIfNeeded()
+        }
+    }
+
+    func parentEditingDidEnd() {
+        if let floatingLabelInput = superview as? FloatingLabelProtocol,
+            floatingLabelInput.isParentEmpty() {
+            UIView.animate(withDuration: 0.1) {
+                self.textColor = self.normalColor
+                self.font = self.font.withSize(self.normalFontSize)
+                self.topConstraint?.constant = (self.superview!.frame.size.height - self.intrinsicContentSize.height) / 2
+                self.superview?.layoutIfNeeded()
+            }
+        } else {
+            self.textColor = self.normalColor
+        }
+    }
+
+    func setTextColor(isParentEditing: Bool) {
```

Comment:
```suggestion
    func updateTextColor(shouldHighlight: Bool) {
```
- `shouldHighlight` is more generic, and makes this code more readable than `isParentEditing`
- `set...` functions should typically take a parameter that will be set, in this case, a `setTextColor` function should take a color parameter. Renaming to `update...` makes more semantic sense.